src/base/base-mixin.js

```diff
@@ -521,10 +523,59 @@ export class BaseMixin {
 
     generateSvg () {
         this._svg = this.root().append('svg');
+    
+        if (this.svgDescription.userSet || this._keyboardAccessible) {
+
+            this._svg.append('desc')
+                .attr('id', `desc-id-${this.__dcFlag__}`)
+                .html(`${this._svgDescription}`);
+
+            this._svg
+                .attr('tabindex', '0')
+                .attr('role', 'img')
+                .attr('aria-labelledby', `desc-id-${this.__dcFlag__}`);
+        }
+
         this.sizeSvg();
         return this._svg;
     }
 
+    /**
+     * Set description text for the entire SVG graphic to be read out by assistive technologies and make
+     * the SVG focusable from keyboard.
+     * @param {String} [description]
+     * @returns {String|BaseMixin}
+     */
+    svgDescription () {
+
+        // overwrite for public API
+        this.svgDescription = description => {
+            this.svgDescription.userSet = true
+            this._svgDescription = description;
+            return this;
+        };
```

Comment:
This seems too complicated. When you call the method, it replaces itself with another method, that, when used as a setter, sets a flag on the function object? And then call it on initialization?

Why not do the following:

* Initialize `_svgDescription` to null
* When `.svgDescription()` is called as a getter, return `_svgDescription || this.constructor.name
* When called as a setter, assign to `_svgDescription` as usual
* When testing if it's been set, use `_svgDescription`, but when getting the value, call `.svgDescription()`

Reply:
Well, this is embarrassing :) Was over-thinking it and saw a problem where there wasn't one, really.